lib/Basics/ReadWriteLock.h

```diff
@@ -64,26 +64,49 @@ class ReadWriteLock {
 
   /// @brief releases the read-lock or write-lock
   void unlock();
-  
+
   /// @brief releases the read-lock
   void unlockRead();
-  
+
   /// @brief releases the write-lock
   void unlockWrite();
 
- private:
-  /// @brief a mutex
-  std::mutex _mut;
+private:
+
+  /// @brief mutex for _readers_bell cv
+  std::mutex _reader_mutex;
```

Comment:
do we actually need two seperate mutexes or would one for both condition_variables be enough ?

Reply:
This is simply a small optimization to avoid the serialization of  concurrent reader/writer threads waiting for the respective condition_variable.